components/mediation-initializer/org.wso2.carbon.mediation.transport.handlers/src/main/java/org/wso2/carbon/mediation/transport/handlers/requestprocessors/swagger/format/SwaggerYamlProcessor.java

```diff
@@ -63,7 +66,11 @@ public void process(CarbonHttpRequest request, CarbonHttpResponse response,
                     responseString =
                             yamlDefinition.dumpAsMap(GSONUtils.gsonJsonObjectToMap(jsonParser.parse(defFromRegistry)));
                 } else {
-                    responseString = yamlDefinition.dumpAsMap(new GenericApiObjectDefinition(api).getDefinitionMap());
+                    AxisConfiguration axisConfiguration =
+                            DataHolder.getInstance().getAxisConfigurationContext().getAxisConfiguration();
+                    ServerConfig serverConfig = new CarbonServerConfig(axisConfiguration);
```

Comment:
Lets get a singleton instance if possible.

Reply:
Since there is already common singleton holding Axis config, we don't need to make this singleton we could use that within CarbonServerConfig. Did improvement in latest commit